climada/entity/exposures/base.py

```diff
@@ -197,28 +177,23 @@ def __init__(self, *args, **kwargs):
             data_crs = data.geometry.crs
         except AttributeError:
             data_crs = None
-        if data_crs and data.crs and data_crs != data.crs:
+        if data_crs and data.crs and not u_coord.equal_crs(data_crs, data.crs):
             raise ValueError("Inconsistent crs definition in data and data.geometry")
 
-        crs = kwargs['crs'] if 'crs' in kwargs \
-              else self.meta['crs'] if 'crs' in self.meta \
-              else data_crs if data_crs \
-              else None
+        crs = (crs if crs is not None
+               else self.meta['crs'] if 'crs' in self.meta
+               else data_crs if data_crs
+               else None)
         if 'crs' in self.meta and not u_coord.equal_crs(self.meta['crs'], crs):
-            raise ValueError("Inconsistent crs definition, crs and meta arguments don't match")
+            raise ValueError("Inconsistent CRS definition, crs and meta arguments don't match")
         if data_crs and not u_coord.equal_crs(data_crs, crs):
-            raise ValueError("Inconsistent crs definition, data doesn't match meta or crs argument")
+            raise ValueError("Inconsistent CRS definition, data doesn't match meta or crs argument")
         if not crs:
             crs = DEF_CRS
-            LOGGER.info('crs set to default value %s', crs)
 
         geometry = kwargs.get('geometry')
         if geometry and isinstance(geometry, str):
             raise ValueError("Exposures is not able to handle customized 'geometry' column names.")
-        if 'crs' in kwargs and not geometry and not 'geometry' in data:
-            kwargs.pop('crs')
-        if 'crs' not in kwargs and (geometry or 'geometry' in data):
-            kwargs['crs'] = crs
```

Comment:
Since `crs` is now an explicit keyword argument, we don't have to deal with it. We just pass it on to `set_gdf` where this will be dealt with.

Reply:
👍